rfcs/0097-no-read-store-dir.md

```diff
@@ -0,0 +1,85 @@
+---
+feature: no-read-store-dir
+start-date: 2021-07-04
+author: Las Safin
+co-authors:
+shepherd-team:
+shepherd-leader:
+related-issues:
+---
+
+# Summary
+[summary]: #summary
+
+Set the permissions for /nix/store to 1771 instead of 1775, disabling reading the directory for other users.
+
+# Motivation
+[motivation]: #motivation
+
+This means that you can not trivially see all the paths in the nix store, e.g. `ls` won't work
+on /nix/store without sudo unless you're in the nixbld group.
+
+Almost everything in NixOS needs access to /nix/store, which means that all your systemd services,
+your flatpak programs, your manually [bubblewrap](https://github.com/containers/bubblewrap)ed programs
+will almost certainly have access to /nix/store and see all of its contents.
+That means they will see your NixOS configuration, initrd secrets (unless your bootloader has support for it),
+all the programs you've built, your usage activity, and possibly also your important secrets if you
```

Comment:
What does `usage activity` mean here?

If it's about discovering which programs a user has installed, wouldn't that still be possible for a malicious containerised app that cannot list `/nix/store` but check the presence of known store paths, by constructing externally the last 10 years of output store paths built by Hydra, and `stat()`ing them on the local machine to check if they exist?

For example, such a program could check whether you have `tor-browser` installed by collecting the ~1000 `*-tor-browser` store paths that Hydra built, and check if you have any of them in the store.

Reply:
You are right, I did not consider that.